x-pack/plugins/security/public/management/users/edit_user/edit_user_page.test.tsx

```diff
@@ -4,228 +4,322 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { act } from '@testing-library/react';
-import { mountWithIntl, nextTick } from '@kbn/test/jest';
-import { EditUserPage } from './edit_user_page';
 import React from 'react';
-import { User, Role } from '../../../../common/model';
-import { ReactWrapper } from 'enzyme';
-import { coreMock, scopedHistoryMock } from '../../../../../../../src/core/public/mocks';
+import {
+  fireEvent,
+  render,
+  waitFor,
+  waitForElementToBeRemoved,
+  within,
+} from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import { coreMock } from '../../../../../../../src/core/public/mocks';
 import { mockAuthenticatedUser } from '../../../../common/model/authenticated_user.mock';
 import { securityMock } from '../../../mocks';
-import { rolesAPIClientMock } from '../../roles/index.mock';
-import { userAPIClientMock } from '../index.mock';
-import { findTestSubject } from '@kbn/test/jest';
-
-const createUser = (username: string, roles = ['idk', 'something']) => {
-  const user: User = {
-    username,
-    full_name: 'my full name',
-    email: 'foo@bar.com',
-    roles,
-    enabled: true,
-  };
-
-  if (username === 'reserved_user') {
-    user.metadata = {
-      _reserved: true,
-    };
-  }
-
-  if (username === 'deprecated_user') {
-    user.metadata = {
-      _reserved: true,
-      _deprecated: true,
-      _deprecated_reason: 'beacuse I said so.',
-    };
-  }
-
-  return user;
+import { Providers } from '../users_management_app';
+import { EditUserPage } from './edit_user_page';
+
+jest.mock('@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
+  htmlIdGenerator: () => () => `id-${Math.random()}`,
+}));
+
+const userMock = {
+  username: 'jdoe',
+  full_name: '',
+  email: '',
+  enabled: true,
+  roles: ['superuser'],
 };
 
-const buildClients = (user: User) => {
-  const apiClient = userAPIClientMock.create();
-  apiClient.getUser.mockResolvedValue(user);
+describe('EditUserPage', () => {
```

Comment:
I like these new tests! Can we add a few tests to verify that error handling works as expected? e.g. failed password change, failed user updates?